File: JULY-2025/21-07-2025/p1957.php
class Solution {

    /**
     * @param String $s
     * @return String
     */
    function makeFancyString($s) {
        
        $isNotConsecutive = true;
        $newString = $s[0];
        
        $loopLimit = strlen($s) + 1;

        // loop and check every word with next letter if they equal, 
        // and also check if previous letter also equal
        // if all 3 of them equal then skip the letter
        // otherwise concat to our newString which is our result

        for ($i = 0; $i < $loopLimit; $i++) {

            if ($s[$i] == $s[$i+1]) {
                
                if($isNotConsecutive) {
                    $newString .= $s[$i+1];
                    $isNotConsecutive = false;
                }
            } else {
                $newString .= $s[$i+1];
                $isNotConsecutive = true;
            }
        }
        return $newString;
    }
}

File: JULY-2025/21-07-2025/p1957.rb
# @param {String} s
# @return {String}
def make_fancy_string(s)
    return if s.length < 1 || s.length > 10**5 || s.ord < 97 || s.ord > 122

    new_str = ''
    c = 0
    for i in 1..(s.length) do
        if s[i - 1] == s[i]
            c = c + 1
        else
            c = 0
        end

        if c >= 2
            next
        else
            new_str << s[i-1]
        end
    end

    new_str
end